jgrapht-core/src/main/java/org/jgrapht/alg/flow/MaximumFlowAlgorithmBase.java

```diff
@@ -60,58 +62,94 @@
      * Default tolerance.
      */
     public static final double DEFAULT_EPSILON = 1e-9;
+    /* tolerance (DEFAULT_EPSILON or user-defined) */
+    public final double epsilon;
 
-    private Extension<V, ? extends VertexExtensionBase> vXs;
-    private Extension<E, ? extends EdgeExtensionBase> eXs;
+    /* input network */
+    protected Graph<V, E> network;
+    /* indicates whether the input graph is directed or not */
+    protected final boolean directed_graph;
 
-    abstract DirectedGraph<V, E> getNetwork();
 
-    <VE extends VertexExtensionBase, EE extends EdgeExtensionBase> void init(
+    protected ExtensionManager<V, ? extends VertexExtensionBase> vertexExtensionManager;
+    protected ExtensionManager<E, ? extends AnnotatedFlowEdge> edgeExtensionManager;
```

Comment:
I still don't get it: why its better than `EdgeExtension`? That sole change blows the whole change-set up, while carrying around 0 profit.